File: online-judge-solutions/Codeforces/802G.py
#https://codeforces.com/problemset/problem/802/G
#Lang		:	Python 3.8
#Time		:	46 ms
#Memory         :	0 KB
s,w=input(),"heidi"
for x in s:
    if(len(w)>0 and w[0]==x): w=w[1:]
print("YES" if len(w)==0 else "NO")

File: online-judge-solutions/Codeforces/802G.cpp
//https://codeforces.com/problemset/problem/802/G
//Lang		:	GNU C++14
//Time		:	15 ms
//Memory	:	0 KB
#include<iostream>
using namespace std;
int main(){
	string s,w;
	while(cin>>s){
		w="heidi";
		for(char x:s){
			if(w[0]==x) w.erase(w.begin());
		}
		cout<<(w.size()==0 ?"YES":"NO")<<"\n";
	}
	return 0;
}
